### PR TITLE
added weighted dice loss

### PR DESCRIPTION
**Pull request by subash-khanal**
This PR introduces a class-weighted `log(cosh(Dice))` loss, which is the modified version of: [this implementation](https://github.com/BloodAxe/pytorch-toolbelt/blob/develop/pytorch_toolbelt/losses/dice.py)

### CODE REVIEW

**Comment by isaaccorley** [comment]
Hi @subash-khanal I think the DiceLoss is basically the same as the implementation in smp which is already a dependency. I recommend we use that with your WeightedDiceLoss and ComboLoss implementations so we don't have to add tests ourselves. https://github.com/qubvel-org/segmentation_models.pytorch/blob/main/segmentation_models_pytorch/losses/dice.py

**Comment by isaaccorley** [comment]
> @isaaccorley the link I used vs. the one you shared appears different, but the implementation looks fairly the same!

That's because the smp implementation is also a modified version of the one you copied from!

File: ftw_tools/torchgeo/losses.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+        intersection = torch.sum(output * target, dim=dims)
+        cardinality = torch.sum(output + target, dim=dims)
+    else:
+        intersection = torch.sum(output * target)
+        cardinality = torch.sum(output + target)
+    dice_score = (2.0 * intersection + smooth) / (cardinality + smooth).clamp_min(eps)
+    return dice_score
+
+
+class DiceLoss(_Loss):
```
[the remark this answers is not in the document]
Comment (subash-khanal) [reply ftw_tools/torchgeo/losses.py]: Thanks! I imported that!

File: ftw_tools/torchgeo/trainers.py [hunk 2]
```diff
@@ -127,6 +127,7 @@ def configure_losses(self) -> None:
         """
         loss: str = self.hparams["loss"]
         ignore_index = self.hparams["ignore_index"]
+        model_kwargs: dict[Any, Any] = self.hparams["model_kwargs"]
```
Comment (isaaccorley) [inline ftw_tools/torchgeo/trainers.py:130]: @subash-khanal why are we adding model_kwargs? These don't seem to be used anywhere in your PR

Comment (subash-khanal) [reply ftw_tools/torchgeo/trainers.py:130]: Sure! We can simply add them.. I have not tested the performance of any weighting scheme on it, so I wanted to retain that flexibility in case people wanted to experiment with that hyperparameter.
